joohyun/models/userDao.js

```diff
@@ -50,7 +50,45 @@ const getUserPosts = async (userId) => {
   }
 };
 
+const userLogin = async (email, password) => {
+  try {
+    const user = await getUserByEmail(email);
+
+    if (!user) {
+      return { status: 401, data: { error: "Invalid email or password" } };
+    }
+
+    const passwordMatch = bcrypt.compareSync(password, user.password);
+
+    if (!passwordMatch) {
+      return { status: 401, data: { error: "Invalid email or password" } };
+    }
+
+    const token = jwt.sign({ userId: user.id }, secretKey, { expiresIn: "1d" });
+
+    return { status: 200, data: { token, message: "Login successful" } };
+  } catch (error) {
+    return { status: 500, data: { error: "Internal Server Error" } };
```

Comment:
상태코드 500은 예외처리를 안 한 것으로 생각 할 수 있습니다. 따라서 400코드를 사용해 주세요
```suggestion
    return { status: 400, data: { error: "Internal Server Error" } };
```